File: js/leetcode/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

 var twoSum = function(nums, target) {

  let hash = {}

  for (let i = 0; i < nums.length; i++) {
    //find a potential number
    let potentialNumber = target - nums[i]

    // if that number exists, then we have our answer
    if (potentialNumber in hash) {
      return [i, hash[i]]
    } else {
      // if not, pop it in here
      hash[nums[i]] = i
    }
  }

}

File: leetcode/go/1-two-sum.go
package leetcode

func twoSum(nums []int, target int) []int {
    
	cache := make(map[int]int)
	
	for i := 0; i < len(nums); i++ {
			
			value, ok := cache[target - nums[i]]
			
			if ok {
					result := []int{i, value}
					return result
			}
			
			cache[nums[i]] = i
	}
	
	result := []int{-1, -1}
	return result
	
}